src/path-symex/var_map.cpp

```diff
@@ -112,26 +112,33 @@ void var_mapt::init(var_infot &var_info)
   }
   else
   {
-    try
+    // Check for the presence of va_args
+    std::size_t found=id2string(var_info.symbol).find("::va_arg");
+    if(found != std::string::npos)
     {
-      const symbolt &symbol=ns.lookup(var_info.symbol);
+      var_info.kind=var_infot::PROCEDURE_LOCAL;
+    }
+    else
+    {
+      try
+      {
+        const symbolt &symbol=ns.lookup(var_info.symbol);
+
+        if(symbol.is_static_lifetime)
+        {
+          if(symbol.is_thread_local)
+            var_info.kind=var_infot::THREAD_LOCAL;
+          else
+            var_info.kind=var_infot::SHARED;
+        } else
+          var_info.kind=var_infot::PROCEDURE_LOCAL;
+      }
```

Comment:
Suggestion: instead of try/catch use

```
const symbolt *symbol=0;
if(ns.lookup(var_info.symbol, symbol))
  throw ...

if(symbol->is_static_lifetime) ...
```